src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -3891,5 +3891,29 @@ void LocalFunction()
             CreateSubmission(source, previous: previous)
                 .VerifyEmitDiagnostics();
         }
+
+        [Fact]
+        public void LocalFunctionsAreNotStatic()
+        {
+            var source = @"
+class C
+{
+    void M()
+    {
+        void LocalFunc()
+        {
+        }
+    }
+}";
+
+            var comp = CreateCompilation(source);
+            var syntaxTree = comp.SyntaxTrees[0];
+            var semanticModel = comp.GetSemanticModel(syntaxTree);
+
+            var localFuncDecl = syntaxTree.GetRoot().DescendantNodes().OfType<LocalFunctionStatementSyntax>().Single();
+            var localFuncSymbol = (IMethodSymbol)semanticModel.GetDeclaredSymbol(localFuncDecl);
+
+            Assert.False(localFuncSymbol.IsStatic);
```

Comment:
maybe it would be nice to have another test verifying the local function isn't static even if it's in a static method?

Reply:
Good suggestion.

@Neme12 I did just want to make sure you saw the `PR For Personal Review Only` label on the side. I want to make sure people don't spend their time reviewing this PR until we decide this is the direction we want to take this API.

Reply:
I see, my bad.